app/Http/Controllers/Auth/OauthLoginController.php

```diff
@@ -21,23 +21,30 @@ public function __construct()
 
     public function login()
     {
-        return Socialite::driver('mediawiki')
-            ->redirect();
+        try {
+            return Socialite::driver('mediawiki')->redirect();
+        } catch (Exception $e) {
+            //Cathing temporary credentials retrival error
+            //We are not concerned about logging this as it's a user issue, easily fixed.
+            abort(400); 
```

Comment:
If *this* is throwing errors, it's a problem with either MediaWiki or the socialite driver package.